src/main/java/com/team/planopedia/API/translators/OpenWeatherMapAPI.java

```diff
@@ -0,0 +1,140 @@
+    //Private number of parameters to be returned
+    private final static int numOfParametersReturned = 2;
+
+    /**
+     * Method used to grab data from the weather API, and return the data into
+     * an Array String
+     *
+     * @param _zipCode
+     * @param _country
+     * @return weatherInf
+     */
+    @Override
+    public String[] getWeather(String _zipCode, String _country) {
+        String[] weatherInf = new String[OpenWeatherMapAPI.numOfParametersReturned];
+        // Build the URL 
+        String urlString = baseURL + callAction + _zipCode + "," + _country + "&appid=" + API_Keys.openWeatherMapAPI();
+        URL url;
+
+        try {
+
+            // Creating the URL and connection
+            url = new URL(urlString);
+            HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+            connection.setRequestMethod("GET");
+
+            // Prints the respone code (USED FOR DEBUGGING)
+            int status = connection.getResponseCode();
+
+            //Response code used for debugging
+            System.out.println("Weather API Response Code: " + status + " ---> SHOWN DURING DEBUG");
+
+            /* Grabs the response from the API and appends it to the content variable
+               until it's empty                                                    */
+            StringBuilder content;
+            try ( BufferedReader inputStream = new BufferedReader(new InputStreamReader(connection.getInputStream()))) {
+                String inputLine;
+                content = new StringBuilder();
+
+                //Appends all of the data to the input line until there is no more data
+                while ((inputLine = inputStream.readLine()) != null) {
+                    content.append(inputLine);
+                }
+            }
+            // Closing the connection after the response has been saved
+            connection.disconnect();
+
+            // Saving the data into an object
+            JSONObject obj = new JSONObject(content.toString());
+
+            // Calls to helper method to the weather information from JSONObjer obj and return as string array
+            weatherInf = OpenWeatherMapAPI.getWeatherfromJSONObj(obj, weatherInf);
+
+        } catch (IOException | JSONException ex) {
+            Logger.getLogger(OpenWeatherMapAPI.class.getName()).log(Level.SEVERE, null, ex);
+        }
+        return weatherInf;
+    }
+
+    /**
+     * Method to extract components for the weather from the JSON Object and to
+     * be put on a String Array
+     *
+     * @param _obj contains information from API to be extracted
+     * @param _WeatherInf the Array String where the weather Information will be
+     * returned
+     * @return _weatherInf the same Array String that is passed is returned with
+     * the weather information
+     */
+    private static String[] getWeatherfromJSONObj(JSONObject _obj, String[] _weatherInf) {
```

Comment:
can we also get the "icon" field from the JSON object, and return the icon url? I looked it up and the icon url is just `https://openweathermap.org/img/w/<icon>.png` where `<icon>` is what you pulled from the json